src/menu/components/boxart.c

```
@@ -29,21 +29,83 @@ component_boxart_t *component_boxart_init (const char *storage_prefix, char *gam
+                return b;
+            }
+        }
     }
+    else { // compatibility mode
+
+        char file_name[8];
+
+        // reset the directory path used for boxart.
+        path = path_init(storage_prefix, BOXART_DIRECTORY);
+
+        sprintf(file_name, "%c%c%c%c.png", game_code[0], game_code[1], game_code[2], game_code[3]);
```

cppcheck

[error] Buffer is accessed out of bounds

(bufferAccessOutOfBounds)